homeassistant/components/netatmo/climate.py

```diff
@@ -100,54 +101,81 @@
 
 SCHEMA_SERVICE_SETSCHEDULE = vol.Schema(
     {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_domain(CLIMATE_DOMAIN),
         vol.Required(ATTR_SCHEDULE_NAME): cv.string,
-        vol.Required(ATTR_HOME_NAME): cv.string,
     }
 )
 
 
 async def async_setup_entry(hass, entry, async_add_entities):
     """Set up the Netatmo energy platform."""
-    auth = hass.data[DOMAIN][entry.entry_id][AUTH]
+    data_handler = hass.data[DOMAIN][entry.entry_id][DATA_HANDLER]
 
-    home_data = HomeData(auth)
+    data_class = "HomeData"
+    await data_handler.register_data_class(data_class)
+    home_data = data_handler.data.get(data_class)
 
-    def get_entities():
+    if not home_data:
+        return
+
+    async def get_entities():
         """Retrieve Netatmo entities."""
         entities = []
-        try:
-            home_data.setup()
-        except pyatmo.NoDevice:
-            return
-        home_ids = home_data.get_all_home_ids()
+
+        def get_all_home_ids():
+            """Get all the home ids returned by NetAtmo API."""
+            if home_data is None:
+                return []
+            return [
+                home_data.homes[home_id]["id"]
+                for home_id in home_data.homes
+                if (
+                    "therm_schedules" in home_data.homes[home_id]
+                    and "modules" in home_data.homes[home_id]
+                )
+            ]
+
+        home_ids = get_all_home_ids()
```

Comment:
```suggestion
        def get_all_home_ids(home_data):
            """Get all the home ids returned by NetAtmo API."""
            if home_data is None:
                return []
            return [
                home_data.homes[home_id]["id"]
                for home_id in home_data.homes
                if (
                    "therm_schedules" in home_data.homes[home_id]
                    and "modules" in home_data.homes[home_id]
                )
            ]

        home_ids = get_all_home_ids(home_data)
```

And then un-nest `get_all_home_ids`